test/fixtures/typescript/ts/transformation/test-unused-import.ts

```diff
@@ -0,0 +1,3 @@
+// @ts-ignore
```

Comment:
@legendecas @kdy1 this test should not pass since
`./node --eval="import { missing } from '.';export { };"`, while in our case the unused import is elided

Reply:
Opened an issue to swc https://github.com/swc-project/swc/issues/9413

Reply:
Does swc require you to pass in `verbatimModuleSyntax` somewhere to make that the case?

Reply:
there is no such flag in the version of swc we are using

Reply:
Hm, yeah. Not sure what "concrete" detection is. https://github.com/swc-project/swc/discussions/6866

Compare to esbuild with the option: https://jakebailey.dev/esbuild-playground/#v3=PTAEAEDMEsBsFMB2BDAtvAXKAxge0TAOYB0AVgM74BQA3laKAERIAuATgJ4AKu0iL5RlgDajYHwAm8AB7EB0xgF0ANPSaRcbVMhZCm8cqkZUAvlSogIMBCnRYBeAtBIVqdBozyoADnHhsAeW8WaHxBLHcGJgA3fwAjHWhUAFlcCQBXBABlDn5kBXs2dPg1MzMLMCg-W0xQSRk5cmkqJO9NFlAaUCzcdAAVDm94UBNQSDZepkp0AFpvZGwAa2RCeEYAbnMZNrYOx3IO6ORYYqwe-sHhgF5QAEYAJgBmABZ1oA

And without: https://jakebailey.dev/esbuild-playground/#v3=PTAEAEDMEsBsFMB2BDAtvAXKAxge0TAOYB0AVgM74BQA3laKAERIAuATgJ4AKu0iL5RlgDajYHwAm8AB7EB0xgF0ANPSaRcbVMhZCm8cqkZUAvlSogIMBCnRYBeAtBIVqdBozyoADnHhsAeW8WaHxBLHcGUEtGADd-ACMdaFQAWVwJAFcEAGUOfmQFezZM+DUzMwswKD9bTFBJGTlyaSoU700WUBpQHNx0ABUOb3hQE1BINn6mSnQAWm9kbABrZEJ4RgBucxkOti7Hci7Y5FhSrD7B4dGAXlAARgAmAGYAFk2gA

Reply:
I guess n1 is correct, we probably should add a list of suggested tsconfig options.
I guess:
```
    "verbatimModuleSyntax": true
    "allowImportingTsExtensions": true
```
What else?

Reply:
Those are definitely the suggested options, but my understanding is that no tsconfig is actually read in the course of this feature, right?

(I suppose there's also useDefineForClass fields, again depending on how wide reaching this new feature is)

Reply:
yeah but we should suggest the users to configure their tsconfig in a certain way to avoid runtime errors

Reply:
`useDefineForClassFields` defaults to true when `target` is ES2022 ([doc](https://www.typescriptlang.org/tsconfig/#useDefineForClassFields)).

I think https://www.typescriptlang.org/docs/handbook/modules/guides/choosing-compiler-options.html#im-compiling-and-running-the-outputs-in-nodejs can be a good start recommendation.

Reply:
> there is no such flag in the version of swc we are using

There is a flag present in the `transform` that is not explicitly stated in the type definition. 
I think it could be included.

https://github.com/swc-project/swc/blob/4b5875099be8b01adeeac9156e01e2b450c0e73a/crates/swc_fast_ts_strip/src/lib.rs#L40-L69

Reply:
> > there is no such flag in the version of swc we are using
> 
> There is a flag present in the `transform` that is not explicitly stated in the type definition. 
> I think it could be included.
> 
> https://github.com/swc-project/swc/blob/4b5875099be8b01adeeac9156e01e2b450c0e73a/crates/swc_fast_ts_strip/src/lib.rs#L40-L69
> https://github.com/magic-akari/swc/blob/4b5875099be8b01adeeac9156e01e2b450c0e73a/crates/swc_ecma_transforms_typescript/src/config.rs#L5-L30

Im not sure I understand, can you provide an example with `swc/wasm-typescript` and `transformSync`

Reply:
> Im not sure I understand, can you provide an example with `swc/wasm-typescript` and `transformSync`

check here:
- https://github.com/swc-project/swc/issues/9413#issuecomment-2278467051
- https://github.com/swc-project/swc/pull/9415

Reply:
@magic-akari thanks it fixes it